### PR TITLE
Verify and fix image platform

### PR DESCRIPTION
**Pull request by ilya-zuyev**
Fixes https://github.com/kubernetes/minikube/issues/10402

This PR add a workaround to fix incorrect manifests in images retrieved from remote registries.

### CODE REVIEW

File: pkg/minikube/image/image.go [hunk 1]
```diff
@@ -172,19 +173,45 @@ func retrieveImage(ref name.Reference) (v1.Image, error) {
...
 		return img, nil
 	}
 
 	klog.Warningf("authn lookup for %+v (trying anon): %+v", ref, err)
-	img, err = remote.Image(ref)
-	return img, err
+	return remote.Image(ref, remote.WithPlatform(p))
+}
+
+func fixPlatform(ref name.Reference, img v1.Image, p v1.Platform) (v1.Image, error) {
```
Comment (medyagh) [inline pkg/minikube/image/image.go:195]: does this hack still need to be there even with latest https://github.com/google/go-containerregistry ?
if yes I recommend creating an issue with example logs to describe it, so this be fixed in the upstream
and add that issue as a comment in this func so we know the exact reason for this hack and also help other projects to fix the bug for everyone

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/image/image.go (hunk 1 above)
Comment (medyagh) [reply pkg/minikube/image/image.go:195]: @ilya-zuyev please add the links to the issue for this hack in the comment

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/image/image.go (hunk 1 above)
Comment (ilya-zuyev) [reply pkg/minikube/image/image.go:197]: Done

File: pkg/minikube/image/image.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package image
 import (
 	"context"
 	"fmt"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
```
Comment (spowelljr) [inline pkg/minikube/image/image.go:22]: Should probably group this import with the rest of the GitHub imports below instead

Comment (ilya-zuyev) [reply pkg/minikube/image/image.go]: good point. fixed

**Review by medyagh — APPROVED** [review]

**Pull request merged**